controllers/application/egress_service_entry.go

```diff
@@ -22,138 +13,35 @@ func (r *ApplicationReconciler) reconcileEgressServiceEntry(ctx context.Context,
-					Endpoints:  endpoints,
-					Ports:      r.getPorts(rule.Ports, rule.Ip, *application),
-				}
-
-				return nil
-			})
+	for _, serviceEntry := range serviceEntries {
+		_, err := ctrlutil.CreateOrPatch(ctx, r.GetClient(), &serviceEntry, func() error {
+			// Set application as owner of the service entry
+			err := ctrlutil.SetControllerReference(application, &serviceEntry, r.GetScheme())
 			if err != nil {
 				r.SetControllerError(ctx, application, controllerName, err)
-				return reconcile.Result{}, err
+				return err
 			}
-		}
-	}
-
-	// Clear out unused service entries
-	serviceEntries := networkingv1beta1.ServiceEntryList{}
-	err := r.GetClient().List(ctx, &serviceEntries, client.InNamespace(application.Namespace))
-	if err != nil {
-		r.SetControllerError(ctx, application, controllerName, err)
-		return reconcile.Result{}, err
-	}
+			r.SetLabelsFromApplication(ctx, &serviceEntry, *application)
+			util.SetCommonAnnotations(&serviceEntry)
 
-	for _, serviceEntry := range serviceEntries.Items {
-		// Skip unrelated service entries
-		if !isEgressServiceEntry(serviceEntry) {
-			continue
-		}
-
-		applicationOwnerIndex := slices.IndexFunc(serviceEntry.GetOwnerReferences(), func(ownerReference metav1.OwnerReference) bool {
-			return ownerReference.Name == application.Name
+			return nil
 		})
-		serviceEntryOwnedByThisApplication := applicationOwnerIndex != -1
-		if !serviceEntryOwnedByThisApplication {
-			continue
-		}
-
-		serviceEntryInApplicationSpecIndex := -1
-
-		if application.Spec.AccessPolicy != nil {
-			serviceEntryInApplicationSpecIndex = slices.IndexFunc(application.Spec.AccessPolicy.Outbound.External, func(rule podtypes.ExternalRule) bool {
-				egressName := fmt.Sprintf("%s-egress-%x", application.Name, util.GenerateHashFromName(rule.Host))
-				return serviceEntry.Name == egressName
-			})
-		}
 
-		ingressGatewayInApplicationSpec := serviceEntryInApplicationSpecIndex != -1
-		if ingressGatewayInApplicationSpec {
-			continue
-		}
-
-		// Delete the rest
-		err = r.GetClient().Delete(ctx, serviceEntry)
-		err = client.IgnoreNotFound(err)
 		if err != nil {
 			r.SetControllerError(ctx, application, controllerName, err)
 			return reconcile.Result{}, err
 		}
 	}
 
-	r.SetControllerFinishedOutcome(ctx, application, controllerName, err)
-
-	return reconcile.Result{}, err
-}
-
-func (r ApplicationReconciler) getPorts(externalPorts []podtypes.ExternalPort, ruleIP string, application skiperatorv1alpha1.Application) []*networkingv1beta1api.Port {
-	ports := []*networkingv1beta1api.Port{}
-
-	if len(externalPorts) == 0 {
-		ports = append(ports, &networkingv1beta1api.Port{
-			Name:     "https",
-			Number:   uint32(443),
-			Protocol: "HTTPS",
-		})
-
-		return ports
-	}
-
-	for _, port := range externalPorts {
-		if ruleIP == "" && port.Protocol == "TCP" {
-			r.GetRecorder().Eventf(
-				&application,
-				corev1.EventTypeWarning, "Invalid",
-				"A static IP must be set for TCP port %d",
-				port.Port,
-			)
```

Comment:
It appears we removed this event from the service_entry.go. Are you confident that we'll be able to debug an issue related to this without it?